reaktive/src/commonTest/kotlin/com/badoo/reaktive/utils/serializer/SerializerTest.kt

```diff
@@ -0,0 +1,171 @@
+package com.badoo.reaktive.utils.serializer
+
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class SerializerTest {
+
+    private val values = arrayListOf<Int>()
+    private val comparator = Comparator<Int>(Int::compareTo)
+
+    @Test
+    fun value_emitted_WHEN_sent() {
+        val serializer =
+            object : Serializer<Int>() {
+                override fun onValue(value: Int): Boolean {
+                    values += value
+                    return true
+                }
+            }
+
+        serializer.accept(1)
+
+        assertValues(1)
+    }
+
+    @Test
+    fun all_values_emitted_in_the_same_order_WHEN_sent_sequentially() {
+        val serializer =
+            object : Serializer<Int>() {
+                override fun onValue(value: Int): Boolean {
+                    values += value
+                    return true
+                }
+            }
+
+        serializer.accept(1)
+        serializer.accept(2)
+        serializer.accept(3)
+
+        assertValues(1, 2, 3)
+    }
+
+    @Test
+    fun comparator_does_not_affect_WHEN_values_sent_sequentially() {
+        val serializer =
+            object : Serializer<Int>(comparator) {
+                override fun onValue(value: Int): Boolean {
+                    values += value
+                    return true
+                }
+            }
+
+        serializer.accept(3)
+        serializer.accept(2)
+        serializer.accept(1)
+
+        assertValues(3, 2, 1)
+    }
+
+    @Test
+    fun value_enqueued_WHEN_sent_from_callback() {
```

Comment:
What do we check with this test? That `onValue` callback was invoked and `success` was set true? We verify callback invocation in previous tests through `assertValues`.
You probably can't test directly, that `2` was added to queue. You can only do it by verifying emitted values like it's done in the next test.

Reply:
I want to verify that when you are submitting a new value while processing another, it doesn't call onValue recursively.

Reply:
And there is a separate test to verify that such a value is processed eventually.

Reply:
I will rename the test

Reply:
@CherryPerry renamed